lib/src/native/realm_core.dart

```diff
@@ -1026,13 +1027,37 @@ class _RealmCore {
     });
   }
 
+  static void _logCallback(Pointer<Void> userdata, int levelAsInt, Pointer<Int8> message) {
+    try {
+      final logger = userdata.toObject<Logger>(isPersistent: true)!;
+      final level = _LogLevel.values[levelAsInt].loggerLevel;
+
+      // Don't do expensive utf8 to utf16 conversion unless we have to..
+      if (logger.isLoggable(level)) {
+        logger.log(level, message.cast<Utf8>().toDartString());
+      }
+    } finally {
+      _realmLib.realm_free(message.cast()); // .. but always free the message
+    }
+  }
+
   SyncClientConfigHandle _createSyncClientConfig(AppConfiguration configuration) {
     return using((arena) {
       final handle = SyncClientConfigHandle._(_realmLib.realm_sync_client_config_new());
 
       _realmLib.realm_sync_client_config_set_base_file_path(handle._pointer, configuration.baseFilePath.path.toUtf8Ptr(arena));
       _realmLib.realm_sync_client_config_set_metadata_mode(handle._pointer, configuration.metadataPersistenceMode.index);
-      _realmLib.realm_sync_client_config_set_log_level(handle._pointer, configuration.logLevel.index);
+      
+      final logger = Realm.logger;
+      _realmLib.realm_sync_client_config_set_log_level(handle._pointer, _LogLevel.fromLevel(logger.level).index);
+      _realmLib.realm_dart_sync_client_config_set_log_callback(
+        handle._pointer,
+        Pointer.fromFunction(_logCallback),
+        logger.toPersistentHandle(),
```

Comment:
Rather than allocating a persistent handle here, we can use `Realm.logger` in the callback.

Reply:
We do.. `logger = Realm.logger`

Reply:
I get it. Yes we can.